### PR TITLE
feat(patreon): link to supporter badge and update integration module

### LINKED ISSUE #3068: Add badge if Patreon Supporter
### Description

We just released the option to connect your Patreon account.  https://github.com/ONEARMY/community-platform/pull/3035
Next we want to display a badge if user is Supporter.

the patreon integration loads data into the user object about their subscription to all onearmy tiers. only users who have integrated with patreon who support the relevant tier (i.e. on PP, only users that support PP or support all one army projects) should get the badge.

Here you can find the icon: https://github.com/ONEARMY/community-platform/blob/master/packages/components/assets/icons/supporter.svg

### Page related

Will be used in :

- User profile page (for group and individual profiles)

### Mockup

afbeelding

### Notes
Came from issue https://github.com/ONEARMY/community-platform/issues/2662

### PR DESCRIPTION
**Pull request by evakill**
PR Checklist

- [x] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Developer experience (improves developer workflows for contributing to the project)

## Description
1. applies the requested changes to the Patreon integration section of the settings page:
image

2. adds the `supporter: true` badge for users that integrate with Patreon and support a linked tier
3. writes tests!

_What this PR does_

## Git Issues

Closes #3068 

## Screenshots/Videos

https://github.com/ONEARMY/community-platform/assets/37253846/83278d65-742f-4f9d-a67b-16250f0d41e1

_If useful, provide screenshot or capture to highlight main changes_

---

## What happens next?

Thanks for the contribution! We try to make sure all PRs are reviewed ahead of our monthly maintainers call (first Monday of the month)

If the PR is working as intended it'll be merged and included in the next platform release, if not changes will be requested and re-reviewed once updated.

If you need more immediate feedback you can try reaching out on Discord in the [Community Platform `development` channel](https://discord.com/channels/586676777334865928/938781727017558018).

### CODE REVIEW

File: functions/src/Integrations/patreon.ts [hunk 1]
```diff
@@ -15,6 +15,48 @@ const PATREON_CLIENT_ID = CONFIG.integrations.patreon_client_id
...
+      // Return true for local development.
+      return true
+  }
+}
+
+export const isSupporter = (patreonUser: PatreonUser) => {
+  const supportsValidTier = patreonUser.membership?.tiers
+    .map(({ id }) => id)
+    .some(isValidTierId)
+  console.log('supportsValidTier', supportsValidTier)
```
Comment (benfurber) [inline functions/src/Integrations/patreon.ts:53]: Did you mean to leave this here?

File: src/pages/UserSettings/content/formSections/PatreonIntegration.tsx [hunk 2]
```diff
@@ -45,11 +71,14 @@ export const PatreonIntegration = ({ user }: { user: IUserPP }) => {
                 height: 'auto',
               }}
             />
-            <Text>Successfully linked patron account!</Text>
+            <Text>Successfully linked Patreon account!</Text>
```
Comment (benfurber) [inline src/pages/UserSettings/content/formSections/PatreonIntegration.tsx:74]: Thoughts on sticking all the text in a label file? I'm trying to do that as standard.

File: src/pages/UserSettings/content/formSections/PatreonIntegration.tsx [hunk 3]
```diff
@@ -99,14 +128,26 @@ export const PatreonIntegration = ({ user }: { user: IUserPP }) => {
...
-        sx={{ width: '40%', justifyContent: 'center' }}
-        variant="outline"
-      >
-        {user.patreon ? 'Sync Patreon Data' : 'Connect To Patreon'}
-      </Button>
+      <Flex>
+        <Button
+          onClick={patreonRedirect}
+          mb={3}
+          sx={{ width: '40%', justifyContent: 'center', mr: 3 }}
```
Comment (benfurber) [inline src/pages/UserSettings/content/formSections/PatreonIntegration.tsx:135]: I think it's worth avoiding things like '40%'. I'd say it's better to flexGrow each item to 1 plus justifyContent of the parent to something like 'space-evenly' and have a large gap (like 4 or something). Then it's all relative.

**Review by benfurber — APPROVED** [review]
I'd prefer if you did makes the changes I've suggested. Not offended if you disagree/choose not to though.

File: functions/src/Integrations/patreon.ts (hunk 1 above)
Comment (evakill) [reply functions/src/Integrations/patreon.ts:53]: nope! thanks

File: src/pages/UserSettings/content/formSections/PatreonIntegration.tsx (hunk 3 above)
Comment (evakill) [reply src/pages/UserSettings/content/formSections/PatreonIntegration.tsx]: done!

**Pull request merged**